### PR TITLE
simulation code about quick_sort and heap_sort

### PR DESCRIPTION
**Pull request by hsi1032**
(Put an `X` inside the `[ ]` to denote check mark `[X]`.)
With issue #315, I make quick sort and heap sort simulations.
However, these things actually not iterative form, but recursive form.
Then, I make simulation as changing is occured
(print after "partition" in quick sort / print after changing parent with child node in heap sort).

Thank you,

- [x] **if done some changes :**
  - [x] wrote short description in the PR explaining what the changes do ?
  - [x] Fixes #[issue number] if related to any issue

### CODE REVIEW

**Comment by hsi1032** [comment]
I don't know why coverage failed, 
I check these simulations runs quite well in my computer

**Comment by danghai** [comment]
@hsi1032 do not worry about coverage. It is fail because the test case does not cover/ run all line codes that you wrote

File: algorithms/sort/heap_sort.py [hunk 1]
```diff
@@ -1,17 +1,21 @@
-def max_heap_sort(arr):
+def max_heap_sort(arr, simulation = False):
```
Comment (goswami-rahul) [inline algorithms/sort/heap_sort.py]: please remove the space around ` = ` in arguments.
```python
def max_heap_sort(arr, simulation=False):
```
Same changes in all function definitions.

Comment (hsi1032) [reply algorithms/sort/heap_sort.py]: I changed it,
Thank you for your comment! :)

**Pull request merged**